Automattic/jetpack
# Donations: Save block content

#### Changes proposed in this Pull Request:

* Adds support for saving the content of a donations block into the post content, so the published view contains the block output.
![Aug-04-2020 19-29-16](https://user-images.githubusercontent.com/1233880/89325281-de32f800-d688-11ea-81d4-a2b382357be3.gif)

* It also follows up https://github.com/Automattic/jetpack/pull/16593#issuecomment-668060633 and implements the content sync logic as suggested there:
  * Content synced across all tabs:
    * Text before fixed amounts (e.g. "Choose an amount").
    * Text before custom amount input (e.g. "Or enter a custom amount").
    * Donate buttons labels (e.g. "Donate"). They have defaults that are specific to tabs ("Donate", "Donate monthly", "Donate yearly"), but changing any of those would override and sync across all tabs.
  * Content not synced across all tabs (each tab has different items):
    * Heading (e.g. "Make a one-time donation").
    * Fixed amounts.
    * Text before donate button (e.g. "Your contribution is appreciated.").

#### Jetpack product discussion

pbMlHh-dW-p2

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* With a WordPress instance of your choice, install Jetpack Beta and activate this branch. Or visit https://jurassic.ninja/create/?jetpack-beta&branch=update/donations-block-save.
* Connect Jetpack to WordPress.com with a paid plan.
* Make sure Jetpack beta blocks are visible by going to /wp-admin/options-general.php?page=companion_settings and activating `JETPACK_BETA_BLOCKS`. 
* Note that if folks are using the sandbox store method (PCYsg-lW4-p2) to also set the `JETPACK__SANDBOX_DOMAIN` value.
* Go to Posts > New and insert a Donations block.
* Edit the block at your pleasure and publish the post.
* Make sure the content is synced across tabs according to https://github.com/Automattic/jetpack/pull/16593#issuecomment-668060633.
* Visit the published post.
* Make sure the block is displayed there and that it looks visually the same as in the editor.
* Make sure only the relevant content for the active tab is displayed.
* Make sure the custom amount input behaves as expected (you can enter an amount, invalid amounts are highlighted with a red border, valid amounts are formatted when the input loses focus).
* Reload the post in the editor.
* Make sure the saved block content is loaded.

#### Proposed changelog entry for your changes:

N/A.
